opencv/opencv.cabal

```diff
@@ -80,13 +75,7 @@ library
 
     ghc-options: -Wall -fwarn-incomplete-patterns -funbox-strict-fields
 
-    if flag(opencv4)
-        cpp-options:      -DOPENCV4
-        ghc-options: -optc-DOPENCV4
-        pkgconfig-depends: opencv4 >= 4.0.0
-    else
-        pkgconfig-depends: opencv >= 3.0.0
-        build-depends: base64-bytestring >= 1.0.0.1
+    pkgconfig-depends: opencv4 >= 4.0.0
```

Comment:
FYI, this doesn't work with cabal2nix:
```
> cabal2nix: post-process: cannot replace name binding Bind (Identifier "opencv") (Path [Identifier "pkgs",Identifier "opencv"]) by Bind (Identifier "opencv3") (Path [Identifier "pkgs",Identifier "opencv3"]) because it's not found in set fromList [Bind (Identifier "opencv4") (Path [Identifier "pkgs",Identifier "opencv4"])]
> CallStack (from HasCallStack):
>   error, called at src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs:211:27 in cabal2nix-2.19.1-24MEcjNppQdH8xbM8xoUph:Distribution.Nixpkgs.Haskell.FromCabal.PostProcess
```
This is the problem:
https://github.com/NixOS/cabal2nix/blob/2099a1f4594f621bb1a2879b793b860aefe4c027/cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L359

Reply:
@htngr That is correct, that hardcode in `cabal2nix` will need to be fixed once this thing here is merged and released.

If you use this branch, you'll have to patch it, or write a manual `cabal2nix` output for now, like this:

```nix
# TODO: Delete this when https://github.com/LumiGuide/haskell-opencv/pull/157 is in nixpkgs
#       and once cabal2nix has removed its hardcode of `opencv3` in
#       https://github.com/NixOS/cabal2nix/blob/0365d9b77086d26ca5197fb48019cedbb0dce5d2/cabal2nix/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L359
{ mkDerivation, aeson, base, base64-bytestring, bindings-DSL
, bytestring, Cabal, containers, criterion, data-default, deepseq
, directory, Glob, haskell-src-exts, inline-c, inline-c-cpp
, JuicyPixels, lens, linear, opencv4, primitive, QuickCheck, repa
, tasty, tasty-hunit, tasty-quickcheck, template-haskell, text
, transformers, vector
, async
, haskell-src-meta
, fetchFromGitHub
, lib
}:
mkDerivation {
  pname = "opencv";
  version = "0.0.2.1";
  src = "${fetchFromGitHub {
    owner = "nh2";
    repo = "haskell-opencv";
    rev = "fe0b7c8813aa090835ee974d90c0436d8c95efff";
    sha256 = "0xsjvkpss2iilb8dnp2w229pwgskv5r6yqrbfdhc9avjj6x4qj6i";
  }}/opencv";
  setupHaskellDepends = [ base Cabal ];
  libraryHaskellDepends = [
    aeson base base64-bytestring bindings-DSL bytestring containers
    data-default deepseq inline-c inline-c-cpp JuicyPixels linear
    primitive repa template-haskell text transformers vector
    async
    haskell-src-meta
  ];
  libraryPkgconfigDepends = [ opencv4 ];
  testHaskellDepends = [
    base bytestring containers data-default directory Glob
    haskell-src-exts JuicyPixels lens linear primitive QuickCheck repa
    tasty tasty-hunit tasty-quickcheck template-haskell text
    transformers vector
  ];
  benchmarkHaskellDepends = [ base bytestring criterion repa ];
  hardeningDisable = [ "bindnow" ];
  description = "Haskell binding to OpenCV-3.x";
  license = lib.licenses.bsd3;
  hydraPlatforms = lib.platforms.none;
  # Just because when installing from git, there's no license file.
  preInstall = ''
    touch LICENSE
  '';
}
```